Makefile

```diff
@@ -1,75 +1,27 @@
-all: init test build
-build: init docs/index.yaml docs/es/all.yaml docs/es/all-latest.yaml
+# Top-level Makefile for working on all helm-charts projects at once.
 
-CHART := enterprise-suite
-export VERSION := $(shell scripts/export-chart-version.sh $(CHART))
-export RELEASE := $(CHART)-$(VERSION)
-CHART_LATEST := $(CHART)-latest
-export RELEASE_LATEST := $(CHART_LATEST)-$(VERSION)
+# Collection of charts to process.
+# These are subdirectories of helm-charts.  Add to the list as required.
+#
+# This setting is the default.  To build a subset (at this level), just define a
+# value on the command line.  e.g. "make CHARTS=sample-project"
+#
+CHARTS = enterprise-suite enterprise-suite-latest reactive-sandbox #sample-project
 
-define banner
-	$(info === $@)
-endef
 
-docs/es/all.yaml: docs/$(RELEASE).tgz
-	$(call banner)
-	helm --namespace=lightbend template $< > $@
+TOPTARGETS := all build clean test lint-helm minikube-test
```

Comment:
I'm not a big fan of sticking everything in `common.mk`, I think it makes it difficult to follow the logic (at least it did for me). At least I'd like if `all` and `build` remain in the top level Makefile since they don't need to be redefined in every sub-Makefile, as far as I can tell. I think `common.mk` should be used only when needed to reduce duplication across the sub-Makefiles.

I also think `lint-helm` is something test specific so shouldn't bleed into the top level here at all (it should be a dependency of `test` in sub-Makefiles).

Also I'm not sure about the `TOPTARGETS` and `MAKECMDGOALS` usage - I get the purpose but for someone who's not a make guru I think it's rather confusing. I'm really allergic to make voodoo, so maybe I'm biased, but I believe simpler the better when it comes to make. Maybe that's a bad reason, but I think if we only have a handful of targets it's better to explicitly spell them out. I think this needs to be understandable/maintainable even by people who are not super familiar with make.

Reply:
Okay I understand the point now - it's because we're trying to template for every single chart in the CHARTS array. Are we making this too complicated? Maybe we should just build a single CHART in the top level Makefile, the way it was before?

Then you can move many of the targets out of `common.mk` into the top level `Makefile` and it removes a lot of the complexity. This fits with travis as well, since there we want to build the charts in parallel so we don't really have the need for a makefile target which builds all charts. We could always write a script which invokes make with each of the charts if we needed to.